src/features/search/pages/Search/Search.tsx

```diff
@@ -67,16 +67,22 @@ export function Search() {
     dispatch({ type: 'SET_STATE', payload: params ?? { view: SearchView.Landing } })
   }, [dispatch, params])
 
-  const currentView = useMemo(() => params?.view, [params?.view])
+  const currentFilters = params?.locationFilter
 
-  const searchVenuePosition = useMemo(
-    () => buildSearchVenuePosition(params?.locationFilter, userPosition),
-    [params?.locationFilter, userPosition]
+  const setQueryHistoryMemoized = useCallback(
+    (query: string) => setQueryHistory(query),
+    [setQueryHistory]
   )
```

Comment:
pourrais-tu m'expliquer la nécessité de ce `useCallback` ?

Reply:
Idem que le useMemo.

Reply:
il se passe quoi si je fais ?
```suggestion
  const setQueryHistoryMemoized = useCallback(
    setQueryHistory,
    [setQueryHistory]
  )
```

et donc

```suggestion
  const setQueryHistoryMemoized = setQueryHistory
```